compute_tools/src/config.rs

```diff
@@ -92,6 +92,25 @@ pub fn write_postgres_conf(
         }
     }
 
+    // Check /proc/sys/vm/overcommit_memory -- if it equals 2 (i.e. linux memory overcommit is
+    // disabled), then the control plane has enabled swap and we should set
+    // dynamic_shared_memory_type = 'mmap'.
+    //
+    // This is (maybe?) temporary - for more, see https://github.com/neondatabase/cloud/issues/12047.
+    let overcommit_memory_contents = std::fs::read_to_string("/proc/sys/vm/overcommit_memory")
+        // ignore any errors - they may be expected to occur under certain situations (e.g. when
+        // not running in Linux).
+        .unwrap_or_else(|_| String::new());
+    if overcommit_memory_contents.trim() == "2" {
```

Comment:
I might might move the trim() to the line above, so that the variable just contains the value and not the newline

Reply:
`trim()` returns `&str` so it can't be done without a separate variable (else it'd be UAF). Is that what you were suggesting? or wdyt